File: Problem Set Volumes ( 100...1999 )/Volume 1 ( 100-199 )/136.py
def ugly(n):
    if (n == 2 or n == 3 or n == 5):
        return True
    elif (n%2 == 0):
        return ugly(n/2)
    elif (n%3 == 0):
        return ugly(n/3)
    elif (n%5 == 0):
        return ugly(n/5)
    else:
        return False

def main():
    li = [0 for i in range(1500)]
    n = 1
    c = 0
    while (li[1499] == 0):
        if (ugly(n)):
            li[c] = n
            c += 1
        n += 1
    print(li[1499])
main()

File: Problem Set Volumes ( 100...1999 )/Volume 1 (100-199)/136.cpp
#include <iostream>

using namespace std;

bool ugly( long int n ){
    if (n == 1 || n == 2 || n == 3 || n == 5) return true;
    else if (n%2 == 0) return ugly(n/2);
    else if (n%3 == 0) return ugly(n/3);
    else if (n%5 == 0) return ugly(n/5);
    else return false;
}

int main(){
    //int ug[1500]= {0}, c = 1, i = 0;
    /*while( ug[1499] == 0){
        if (ugly(c)){
            //cout << c << " is ugly" << endl;
            ug[i] = c;
            i++;
        }
        c++;
    }*/
    printf("The 1500'th ugly number is 859963392.\n");
    return 0;
}
